python/ray/worker.py

```diff
@@ -891,6 +887,71 @@ def custom_excepthook(type, value, tb):
 UNCAUGHT_ERROR_GRACE_PERIOD = 5
 
 
+def _set_log_file(file_name, worker_pid, old_obj, setter_func):
+    # Line-buffer the output (mode 1).
+    f = open_worker_log(file_name, worker_pid)
+
+    # Before redirecting stdout we need to make sure userspace buffers
+    # are all flushed.
+    # Flush python's userspace buffers
+    old_obj.flush()
```

Comment:
Have you run into any issues without flushing? Consider removing this.

Reply:
I can't find any documentation or guarantee's on python's buffer flushing policy so I'll remove this line but leave a comment. Python does seem to always flush, but this feels like we're relying on very sketchy undefined behavior.